File: solutions/typescript/secret-handshake/1/secret-handshake.ts
const WINK = 1;
const DOUBLE_BLINK = 2;
const CLOSE_EYES = 4;
const JUMP = 8;
const REVERSE = 16;

export function commands(n: number): string[] {
  const output = [];
  if(WINK & n) {
    output.push("wink");
  }
  if(DOUBLE_BLINK & n) {
    output.push("double blink");
  }
  if(CLOSE_EYES & n) {
    output.push("close your eyes");
  }
  if(JUMP & n) {
    output.push("jump");
  }
  if(REVERSE & n) {
    return output.reverse();
  } else {
    return output;
  }
}

File: solutions/python/secret-handshake/1/secret_handshake.py
WINK = 1
DOUBLE_BLINK = 2
CLOSE_EYES = 4
JUMP = 8
REVERSE = 16

def commands(binary_str):
    number = int(binary_str, 2)
    output = list()
    if number & WINK:
        output.append("wink")
    if number & DOUBLE_BLINK:
        output.append("double blink")
    if number & CLOSE_EYES:
        output.append("close your eyes")
    if number & JUMP:
        output.append("jump")
    if number & REVERSE:
        output.reverse()
    return output
